core/templates/pages/header_js_libs.html

```diff
@@ -26,5 +26,6 @@
 <% } else if (htmlWebpackPlugin.options.hybrid) { %>
   <script src="/third_party/static/angularjs-1.8.2/angular.js"></script>
 <% } %>
-<!-- TODO(#17664): Fetch widget.js from donorbox only in the donate page. -->
-<script async src="https://donorbox.org/widget.js" paypalExpress="false"></script>
+<% if (typeof window !== 'undefined' && window.location.href.includes('donate')) { %>
+  <script async src="https://donorbox.org/widget.js" paypalExpress="false"></script>
+<% } %>
```

Comment:
Can you move this to the angular side of code instead? I would say we can create a script loading service. Here is one for you:
```
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class ScriptLoaderService {

  private scripts: any = {};

  load(scriptUrl: string): Promise<any> {
    return new Promise((resolve, reject) => {
      // check if script already exists
      if (this.scripts[scriptUrl]) {
        resolve(this.scripts[scriptUrl]);
      } else {
        // load the script
        const scriptElement = document.createElement('script');
        scriptElement.src = scriptUrl;
        scriptElement.onload = () => {
          this.scripts[scriptUrl] = { loaded: true, src: scriptUrl };
          resolve(this.scripts[scriptUrl]);
        };
        scriptElement.onerror = (error: any) => reject(error);
        document.getElementsByTagName('body')[0].appendChild(scriptElement);
      }
    });
  }
}
```

You can use this in the donate page to load it dynamically rather than using tenuous logic in the html file. 
If you're interested in proceeding with the current approach, could you demonstrate in the 'Proof of Correct Changes' section that the script is only loading on the 'Donate' page and not on the 'About' page? You could provide evidence by capturing a screenshot of the Developer Console's Network tab, with the JS/CSS sub-tab activated.

Reply:
done.